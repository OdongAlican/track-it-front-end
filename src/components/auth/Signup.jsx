/* eslint-disable react/require-default-props */

import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { signUpAction } from '../../actions';

const Signup = props => {
  const { history } = props;
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [age, setAge] = useState('');
  const dispatch = useDispatch();

  const submitData = event => {
    event.preventDefault();
    dispatch(signUpAction({ username, password, age }, history));
  };

  const errorMessage = useSelector(state => state.authReducer.error);

  return (
    <div data-testid="appSignup" className="general-login-section">
      <div className="login-section p-3">
        <h3>Sign Up</h3>
        <form onSubmit={submitData}>
          <input
            placeholder="Enter User Name"
            value={username}
            onChange={e => setUsername(e.target.value)}
          />
          <input
            placeholder="Enter Password"
            type="password"
            value={password}
            onChange={e => setPassword(e.target.value)}
          />
          <input
            placeholder="Enter Age"
            value={age}
            onChange={e => setAge(e.target.value)}
          />
          <p className="text-danger">{ errorMessage || '' }</p>
          <button type="submit" className="btn btn-primary">Submit</button>
        </form>
        <Link to={{
          pathname: '/',
        }}
        >
          Already have an Account?
        </Link>
      </div>

    </div>
  );
};

Signup.propTypes = {
  history: PropTypes.string,
};

export default Signup;
