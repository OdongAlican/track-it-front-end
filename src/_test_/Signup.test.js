import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Signup from '../components/auth/Signup';

afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};

const mockStore = configureStore();
let store;
const props = {
  history: '/',
};

it('renders Signup correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Signup props={props} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appSignup');
  expect(result[0]).toHaveTextContent('Sign UptestSubmitAlready have an Account?');
});

it('renders Signup correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Signup props={props} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appSignup');
  expect(result[0]).not.toHaveTextContent('Some Random Text');
});

it('matches Signup snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><Signup props={props} /></Provider>
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
