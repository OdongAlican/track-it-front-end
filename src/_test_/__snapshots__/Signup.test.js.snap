// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Signup snapshot 1`] = `
<div
  className="general-login-section"
  data-testid="appSignup"
>
  <div
    className="login-section p-3"
  >
    <h3>
      Sign Up
    </h3>
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Enter User Name"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="Enter Age"
        value=""
      />
      <p
        className="text-danger"
      >
        test
      </p>
      <button
        className="btn btn-primary"
        type="submit"
      >
        Submit
      </button>
    </form>
    <a
      href="/"
      onClick={[Function]}
    >
      Already have an Account?
    </a>
  </div>
</div>
`;
